clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -296,9 +281,11 @@ public boolean awaitPendingRequests(Node node, long timeoutMs) {
      */
     public int pendingRequestCount(Node node) {
         synchronized (this) {
-            List<ClientRequest> pending = unsent.get(node);
-            int unsentCount = pending == null ? 0 : pending.size();
-            return unsentCount + client.inFlightRequestCount(node.idString());
+            int total = 0;
+            for (UnsentRequest request : unsent)
```

Comment:
Yes, `hasPendingRequests(node)` would still be worst-case O(n), but `hasPendingRequests()` would provide more benefit, right? If you don't want to do it, I will just do it in a follow-up. I was trying to address your apparent concerns about performance. And is difficulty of unit tests really an issue here? You can trivially add to existing tests.